gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -181,6 +184,34 @@ private ManagedChannel createSingleChannel() throws IOException {
     int port = Integer.parseInt(endpoint.substring(colon + 1));
     String serviceAddress = endpoint.substring(0, colon);
 
+    // TODO(hzyi): Use NettyChannelBuilder when maxHeaderListSize is specified to unblock Spanner.
+    // Changed to ManagedChannelBuilder when https://github.com/grpc/grpc-java/issues/4050 is resolved.
+    if (maxHeaderListSize != null) {
+      NettyChannelBuilder builder =
+          NettyChannelBuilder.forAddress(serviceAddress, port)
```

Comment:
`NettyChannelBuilder` is-a `ManagedChannelBuilder`, so in this block you only need to use `forAddress` followed by `maxHeaderListSize`, then the rest of this can be shared code.